### PR TITLE
[ZEPPELIN-5964] Fix renaming bug

### PR DESCRIPTION
**Pull request by matthias-koch**
### What is this PR for?
If a note is renamed by changing only the capitalization, the note is copied at the file system level when it is executed again.

### What type of PR is it?
Bug Fix

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-5964

### How should this be tested?
* New unit test were added

### Questions:
* Does the license files need to update?
* Is there breaking changes for older versions?
* Does this needs documentation?

### CODE REVIEW

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java [hunk 1]
```diff
@@ -247,7 +247,7 @@ public void moveNote(String noteId,
     this.notebookRepo.move(noteId, notePath, newNotePath, subject);
 
     // Update path of the note
-    if (!StringUtils.equalsIgnoreCase(notePath, newNotePath)) {
+    if (!notePath.equals(newNotePath)) {
```
Comment (Reamer) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java:250]: `StringUtils.equals(x, y)` is null safe and should be preferred.

**Review by Reamer — COMMENTED** [review]
Except for one small comment, I think the PR is very good.
I don't know if all notebook storage systems are case sensitive. This could lead to problems. Could you check this?

**Review by jongyoul — APPROVED** [review]
I agreed with @Reamer's concern but basically, it's good

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java (hunk 1 above)
Comment (matthias-koch) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java]: Git: case sensitive
Hadoop: case sensitive
S3: case sensitive
Azure: case sensitive
Google Cloud Storage: case sensitive
Aliyun OSS: case sensitive
MongoDB: case sensitive
Github: case sensitive

It seems we are safe.

Comment (Reamer) [reply zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java]: awesome

**Review by Reamer — APPROVED** [review]
LGTM

**Pull request merged**